README.md

```diff
@@ -128,17 +148,222 @@ To start Parity Ethereum as a regular user using `systemd` init:
+/// // insert example 1 code here
+/// ```
+///
+```
+
+* Important notes:
+  * Documentation comments must use annotations with a triple slash `///`
+  * Modules are documented using `//!`
+```
+//! Summary (of module)
+//!
+//! Description (of module)
+```
+* Special section header is indicated with a hash `#`.
+  * `Panics` section requires an explanation if the function triggers a panic
+  * `Errors` section is for describing conditions under which a function of method returns `Err(E)` if it returns a `Result<T, E>`
+  * `Safety` section requires an explanation if the function is `unsafe`
+  * `Examples` section includes examples of using the function or method
+* Code block annotations for examples are included between triple graves, as shown above.
+Instead of including the programming language to use for syntax highlighting as the annotation
+after the triple graves, alternative annotations include the `ignore`, `text`, `should_panic`, or `no_run`.
+* Summary sentence is a short high level single sentence of its functionality
+* Description paragraph is for details additional to the summary sentence
+* Missing documentation annotations may be used to identify where to generate warnings with `#![warn(missing_docs)]`
+or errors `#![deny(missing_docs)]`
+* Hide documentation for items with `#[doc(hidden)]`
+
+### Contributing to documentation (tests, extended examples, macros) for Parity Ethereum packages
+
+The code block annotations in the `# Example` section may be used as [documentation as tests and for extended examples](https://doc.rust-lang.org/1.9.0/book/documentation.html#documentation-as-tests).
+
+* Important notes:
+  * Rustdoc will automatically add a `main()` wrapper around the code block to test it
+  * [Documenting macros](https://doc.rust-lang.org/1.9.0/book/documentation.html#documenting-macros).
+  * Documentation as tests examples are included when running `cargo test`
+
+## 5. Toolchain <a id="chapter-005"></a>
 
 In addition to the Parity Ethereum client, there are additional tools in this repository available:
 
-- [evmbin](https://github.com/paritytech/parity-ethereum/blob/master/evmbin/) - EVM implementation for Parity Ethereum.
-- [ethabi](https://github.com/paritytech/ethabi) - Parity Ethereum function calls encoding.
-- [ethstore](https://github.com/paritytech/parity-ethereum/blob/master/accounts/ethstore) - Parity Ethereum key management.
-- [ethkey](https://github.com/paritytech/parity-ethereum/blob/master/accounts/ethkey) - Parity Ethereum keys generator.
-- [whisper](https://github.com/paritytech/parity-ethereum/blob/master/whisper/) - Implementation of Whisper-v2 PoC.
+- [evmbin](https://github.com/paritytech/parity-ethereum/blob/master/evmbin/) - Parity Ethereum EVM Implementation.
+- [ethabi](https://github.com/paritytech/ethabi) - Parity Ethereum Encoding of Function Calls.
+- [ethstore](https://github.com/paritytech/parity-ethereum/blob/master/accounts/ethstore) - Parity Ethereum Key Management.
```

Comment:
IMHO we should use relative paths to projects within the repo. I'd also group tools from external repos together.

Reply:
I've addressed this in these commits 55d1a3e2cf677a4c37a84d19bf3e6972ff5c6d57, d880c01081c4279599332555f72c10fffd23cff1